### PR TITLE
[rosa create cluster] Verify provided subnets for Existing VPC exist in AWS

### PR DESCRIPTION
**Pull request by nimrodshn**
CC: @vkareh  @igoihman

### CODE REVIEW

File: cmd/create/cluster/cmd.go [hunk 1]
```diff
@@ -428,9 +444,7 @@ func run(cmd *cobra.Command, _ []string) {
 			options[i] = setSubnetOption(subnetID, availabilityZone)
 			if subnetsProvided {
 				for _, subnetArg := range subnetIDs {
-					if subnetArg == subnetID {
-						optionDefaults = append(optionDefaults, setSubnetOption(subnetID, availabilityZone))
-					}
+					defaultOptions = append(defaultOptions, setSubnetOption(subnetArg, availabilityZone))
```
Comment (vkareh) [inline cmd/create/cluster/cmd.go:447]: Doesn't this make it so that _all_ subnets are selected, regardless of which were sent in the CLI?

Comment (nimrodshn) [reply cmd/create/cluster/cmd.go:447]: @vkareh why? only those passed via the CLI (i.e. `subnetArg`) are selected..

Comment (vkareh) [reply cmd/create/cluster/cmd.go:447]: duh, makes sense :stuck_out_tongue_closed_eyes:

File: cmd/create/cluster/cmd.go [hunk 2]
```diff
@@ -418,7 +418,23 @@ func run(cmd *cobra.Command, _ []string) {
...
+
+		// Verify subnets provided exist.
+		if subnetsProvided {
+			for _, subnetArg := range subnetIDs {
+				verifiedSubnet := false
+				for _, subnet := range subnets {
+					if awssdk.StringValue(subnet.SubnetId) == subnetArg {
+						verifiedSubnet = true
+					}
+				}
```
Comment (vkareh) [inline cmd/create/cluster/cmd.go:431]: Can we do this check inside the next loop? We already have the same loop doing
```
if subnetsProvided {
  for _, subnetArg := range subnetIDs {
// ...
```
It's the exact same loop, let's reuse it!

Comment (nimrodshn) [reply cmd/create/cluster/cmd.go:431]: I wanted to error early..

**Pull request merged**